### PR TITLE
add SearchBar and its container

### PR DESCRIPTION
**Pull request by harunurhan**

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-import reducer from './reducer';
+import reducers from './reducers';
 
 export const history = createHistory();
 const reduxRouterMiddleware = routerMiddleware(history);
 
 const getMiddleware = () => {
   if (process.env.NODE_ENV === 'production') {
-    return applyMiddleware(reduxRouterMiddleware);
+    return applyMiddleware(reduxRouterMiddleware, thunk);
```
Comment (drjova) [inline src/store.js]:
```js
import http from 'common/http';

return applyMiddleware(
    reduxRouterMiddleware,
    thunk.withExtraArgument(http)
)
```

File: src/actions/search.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+function searchError(error) {
+  return {
+    type: types.SEARCH_ERROR,
+    payload: error,
+  };
+}
+
+export default function search(query) {
+  return async (dispatch) => {
```
Comment (drjova) [inline src/actions/search.js]:
```js
return async (dispatch, getState, api) => {

```

**Comment by harunurhan** [comment]
@drjova shall I add `immutablejs` as well?

**Comment by harunurhan** [comment]
@drjova one more thing to decide is that, should action.payload be immutablejs or we convert it if needed in reducer.

**Pull request merged**